tools/tcat_ble_client/cli/base_commands.py

```diff
@@ -105,6 +105,68 @@ async def execute_default(self, args, context):
         return CommandResultTLV(tlv_response)
 
 
+class GetDeviceIdCommand(Command):
+
+    def get_help_string(self) -> str:
+        return 'Get unique identifier for the TCAT device.'
+
+    async def execute_default(self, args, context):
+        bless: BleStreamSecure = context['ble_sstream']
+        data = (TLV(TcatTLVType.GET_DEVICE_ID.value, bytes()).to_bytes())
+        response = await bless.send_with_resp(data)
+        if not response:
+            return
+        tlv_response = TLV.from_bytes(response)
+        return CommandResultTLV(tlv_response)
```

Comment:
I'm wondering if we could create a helper function for sending TLVs which could take context and data as arguments? I can see this code is repeated quite few times while only thing that differs is TLV type (and sometimes prints, but as the response should be returned, they could still be added after calling the helper function)

Reply:
Another question on these (repeated) code blocks:  how is a response-with-status with error handled? If the expected payload is not sent but an error; I don't see code for handling this error case. 
For example it looks like if a network name payload is not returned, the code `print(f'Network name: "{str(tlv_response.value)}".')` for printing the  network name is still executed. (But maybe I'm wrong here.) It should print some explanation of the error that occurred , I think.  I.e. print both the error code/number and the human-readable explanation of what that is.  (Maybe I missed this - if so sorry!)

Reply:
@maciejbaczmanski @EskoDijk Please revise newer version of this.